File: Courzelo-backend-LMS/Courzelo/src/main/java/org/example/courzelo/models/ProjectEntities/publication/Publication.java
package org.example.courzelo.models.ProjectEntities.publication;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.example.courzelo.models.ProjectEntities.project.Project;
import org.example.courzelo.models.User;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "Publication")
public class Publication {

    @Id
    private String id;
    @Indexed
    private String content;
    @Indexed
    private LocalDateTime dateTime;
    @Indexed
    private int likes;
    @Indexed
    private int dislikes;
    @Indexed
    private Map<String, String> userReactions;
    @Indexed
    private int commentsCount;
    @DBRef
    private List<Comment> comments;
    @DBRef
    private User author;
    @DBRef
    private Project project;
}

File: courzelo-dashboard-LMS-main/src/app/shared/models/Project/Publication.ts

import { UserResponse } from "../user/UserResponse";
import { Commment } from "./Commment";
import { Reaction } from "./Reaction";




export interface Publication {
    id: string;
    author: UserResponse;
    userReactions: { [key: string]: string };
    dateTime: Date;
    content: string;
    likes: number;
    dislikes: number;
    commentsCount: number;
    comments: Commment[];
    projectId: string; 
}